ue4docker/infrastructure/WindowsUtils.py

```diff
@@ -7,18 +7,21 @@
 
 class WindowsUtils(object):
 
+	# The oldest Windows build we support
+	_minimumRequiredBuild = 17763
+
 	# The latest Windows build version we recognise as a non-Insider build
 	_latestReleaseBuild = 19042
 
 	# The list of Windows Server Core base image tags that we recognise, in ascending version number order
-	_validTags = ['ltsc2016', '1709', '1803', 'ltsc2019', '1903', '1909', '2004', '20H2']
+	_validTags = ['1809', '1903', '1909', '2004', '20H2']
```

Comment:
I think this change (`ltsc2019`->`1809` specifically) is the wrong direction. `ltsc2022` images don't have an equivalent SAC release tag, and it would be more consistent to refer to the LTSC releases by their official `ltscXXXX` tag, so we'll need to bring back management for that later anyway. I'm not sure what the `DisplayVersion` on Server LTSC 2022 is, it might be 21H2 (there was some references to that elsewhere) but the images produced by ue4-docker _should_ still say `ltsc2019` or `ltsc2022` respectively, since "1809" is weird if you don't realise that's "Server 2019", and _any_ SAC version for LTSC 2022 won't actually exist as a tag, or anywhere else in the collective consciousness of the users.

Reply:
Uh... Okay. I just wanted to avoid ths back-and-forth mapping because there isn't `mcr.microsoft.com/windows:ltsc2019`.

Reply:
I remembered, the same-kernel release of Azure Stack HCI is 21H2, that's where I saw that numbering. It's still in Preview, so perhaps when it ships, perhaps there'll be 21H2-tagged container images.

However, I _suspect_ that the intended direction is for Azure Stack HCI to host LTSC 2022 VMs, and then _those_ host LTSC 2022-based containers orchestrated with Kubernetes. The Azure Stack HCI documentation is _really_ lean on such details though, so I'm guessing heavily there.

Reply:
Yeah, lack of `mcr.microsoft.com/windows:ltsc2019` is annoying, I hadn't noticed that.

I think we'll have to manually handle that, because it's a unique case (the other SAC/Client versions we support do have equivalents there), and going forward, no new tags will be added to that repo anyway; instead we'll need to support `mcr.microsoft.com/windows/server` which _only_ has `ltsc2022` and kernel-version tags, and I expect will never have SAC-style tags.

Reply:
> our target audience won't know what 21H2 is, or worse, will confuse it with Windows 10 21H2

I'm not sure I understand what you mean here. ltsc2022 identifies itself as 21H2, and MS clearly says that: https://docs.microsoft.com/en-us/windows-server/get-started/windows-server-release-info#windows-server-current-versions-by-servicing-option

Reply:
Ah, I hadn't noticed they were using 21H2 in the official docs. They're not using it in the Docker tags though, and I _still_ think it's confusing given that [Windows 10 21H2 also exists](https://blogs.windows.com/windowsexperience/2021/07/15/introducing-the-next-feature-update-to-windows-10-21h2/), and is running a different kernel version. Apparently [Windows 11 previews are _also_ 21H2](https://techcommunity.microsoft.com/t5/windows-hardware-certification/now-accepting-windows-11-version-21h2-submissions/ba-p/2483049), so it's pretty much out-the-window as a shorthand for any particular Kernel Build, which is what we actually care about in container-land.

And for ue4-docker, if they are using 21H2 as `DisplayVersion`, we'll need to check elsewhere in that key to distinguish Windows Server 2022 from Windows 10 21H2, as we can only support the former with process isolation, barring some kind of unexpected turnaround on current OS container image plans.

The SAC version numbers are a neat shorthand, but I think using them for releases without a kernel-compatible SAC release is probably a mistake on MS's part, and going to lead to confusion over time. I assume this is why they're not using `21h2` as a container image tag, and `ltsc2022` is now primary (based on it having the KB variants attached, while `ltsc2019` did not).

I kind-of hope they "disappear" `21H2` in reference to Windows Sever 2022 pretty quickly. I still think that long-term, we will need to drop the SAC version names in favour of matching the actual build numbers, but I recall that didn't shake out well last time you tried it.

Reply:
> Windows 10 21H2 also exists, and is running a different kernel version. Apparently Windows 11 will also be 21H2

Oh my god, what a mess.

> we will need to drop the SAC version names and ltsc floating tags in favour of matching the actual build numbers, but I recall that didn't shake out well last time you tried it.

Yep, I didn't like what it turned into. Possibly needs another attempt given that we now have more info about future Windows versioning.

Reply:
See https://github.com/adamrehn/ue4-docker/pull/191